suit_generator/cmd_cache_create.py

```diff
@@ -6,116 +6,297 @@
+    cmd_cache_create_from_payloads.add_argument(
         "--input",
         required=True,
         action="append",
         help="Input binary with corresponding URI, passed in format <URI>,<INPUT_FILE_PATH>."
         + "Multiple inputs can be passed.",
     )
-    cmd_cache_create_arg_parser.add_argument("--output-file", required=True, help="Output raw SUIT DFU cache file.")
-    cmd_cache_create_arg_parser.add_argument(
+
+    cmd_cache_create_from_envelope = cmd_cache_create_subparsers.add_parser(
+        CACHE_CREATE_FROM_ENVELOPE_CMD, help="Create a cache partition from the payloads inside the provided envelope.")
+
+    cmd_cache_create_from_envelope.add_argument("--output-file", required=True, help="Output raw SUIT DFU cache file.")
+    cmd_cache_create_from_envelope.add_argument(
         "--eb-size", type=int, help="Erase block size in bytes (used for padding).", default=16
     )
 
+    cmd_cache_create_from_envelope.add_argument(
+        "--input-envelope",
+        required=True,
+        help="Input envelope file path."
+    )
 
-def add_padding(data: bytes, eb_size: int) -> bytes:
-    """
-    Add padding to the given data to align it to the specified erase block size.
+    cmd_cache_create_from_envelope.add_argument(
+        "--output-envelope",
+        required=True,
+        help="Output envelope file path (envelope with removed extracted payloads)."
+    )
 
-    This function ensures that the data is padded to a size that is a multiple of the erase block size (eb_size).
-    The padding is done by appending a CBOR key-value pair with empty URI as the key and
-    byte-string-encoded zeros as the value.
+    cmd_cache_create_from_envelope.add_argument(
+        "--omit-payload-regex",
+        help="Integrated payloads matching the regular expression will not be extracted to the cache."
+    )
 
-    :param data: The input data to be padded.
-    :type data: bytes
-    :param eb_size: The erase block size in bytes.
-    :type eb_size: int
-    :return: The padded data.
-    """
-    rounded_up_size = math.ceil(len(data) / eb_size) * eb_size
-    padding_size = rounded_up_size - len(data)
-    padded_data = data
-
-    # minimum padding size is 2 bytes
-    if padding_size == 1:
-        padding_size += eb_size
-        rounded_up_size += eb_size
-
-    if padding_size == 0:
-        return data
-
-    padded_data += bytes([0x60])
-
-    if padding_size <= 23:
-        header_len = 2
-        padded_data += bytes([0x40 + (padding_size - header_len)])
-    elif padding_size <= 0xFFFF:
-        header_len = 4
-        padded_data += bytes([0x59]) + (padding_size - header_len).to_bytes(2, byteorder="big")
-    else:
-        raise ValueError("Number of required padding bytes exceeds assumed max size 0xFFFF")
+    cmd_cache_create_from_envelope.add_argument(
+        "--dependency-regex",
+        help="Integrated payloads matching the regular expression will be treated as dependency envelopes and parsed hierarchically."
```

Comment:
nit: Missing space will result in following output:
```
  --dependency-regex DEPENDENCY_REGEX
                        Integrated payloads matching the regular expression will be treated as dependency envelopes and parsed hierarchically.The
                        payloads extracted from the dependency envelopes will be added to the cache.
```

Reply:
Fixed